lib/changelog/oban_workers/stats_processor.ex

```diff
@@ -1,59 +1,57 @@
-defmodule Changelog.Stats do
-  import Ecto
-  import Ecto.Changeset
+defmodule Changelog.ObanWorkers.StatsProcessor do
+  use Oban.Worker, queue: :scheduled, unique: [period: 600]
+
+  import Ecto.Query, only: [select: 3]
 
   alias Changelog.{Cache, Podcast, Repo, Episode, EpisodeStat}
   alias Changelog.Stats.{Analyzer, Parser, S3}
+  alias Ecto.Changeset
 
   require Logger
 
-  def process do
-    end_date = Timex.today()
-    start_date = Timex.shift(end_date, days: -2)
+  @impl Oban.Worker
+  def perform(%Job{args: %{"date" => date, "podcast_id" => podcast_id}}) do
+    date = Date.from_iso8601!(date)
+    podcast = Repo.get!(Podcast, podcast_id)
 
-    for time <- Timex.Interval.new(from: start_date, until: end_date) do
-      time |> Timex.to_date() |> process()
-    end
-  end
-
-  def process(date) do
-    Logger.info("Stats: Start processing for #{date}")
-    podcasts = Repo.all(Podcast.public())
-    process(date, podcasts)
-    Cache.delete_prefix("stats-")
-    Logger.info("Stats: Finished processing for #{date}")
-  end
+    processed =
+      date
+      |> S3.get_logs(podcast.slug)
+      |> Parser.parse()
+      |> Enum.group_by(& &1.episode)
+      |> Enum.map(fn {slug, entries} -> process_episode(date, podcast, slug, entries) end)
 
-  def process(date, podcast) when not is_list(podcast), do: process(date, [podcast])
+    Podcast.update_stat_counts(podcast)
 
-  def process(date, podcasts) do
-    podcasts
-    |> Enum.map(&Task.async(fn -> process_podcast(date, &1) end))
-    # 10 minutes
-    |> Enum.flat_map(&Task.await(&1, 600_000))
+    {:ok, processed}
   end
 
-  defp process_podcast(date, podcast) do
-    Logger.info("Stats: Processing #{podcast.name}")
+  def perform(_job) do
+    today = Date.utc_today()
+    range = Date.range(Date.add(today, -2), Date.add(today, -1))
```

Comment:
Newer versions of Elixir have native Date range iteration, which allows us to skip the Timex interval with date conversion step. Fun fact, the use of `-2` and `-1` is because Timex and Date ranges differed by one day.